packages/playground/website/src/components/playground-viewport/index.tsx

```diff
@@ -21,52 +28,110 @@ interface PlaygroundViewportProps {
 	className?: string;
 }
 
-export const PlaygroundViewport = React.forwardRef<
-	HTMLIFrameElement,
-	PlaygroundViewportProps
->(
-	(
-		{ displayMode = 'browser', toolbarButtons, hideToolbar, className },
-		ref
-	) => {
-		const { playground, currentUrl: url } = usePlaygroundContext();
+export const PlaygroundViewport = ({
+	displayMode = 'browser-full-screen',
+	toolbarButtons,
+	hideToolbar,
+	className,
+}: PlaygroundViewportProps) => {
+	const clientInfo = useAppSelector(getActiveClient);
+	const playground = clientInfo?.client;
+	const url = clientInfo?.url;
 
-		if (displayMode === 'seamless') {
-			// No need to boot the playground if seamless.
-			return <JustViewport iframeRef={ref} />;
-		}
-		return (
-			<BrowserChrome
-				showAddressBar={!!playground}
-				url={url}
-				toolbarButtons={toolbarButtons}
-				onUrlChange={(url) => playground?.goTo(url)}
-				hideToolbar={hideToolbar}
-				className={className}
-			>
-				<JustViewport iframeRef={ref} />
-			</BrowserChrome>
-		);
+	if (displayMode === 'seamless') {
+		// No need to boot the playground if seamless.
+		return <JustViewport />;
 	}
-);
+	return (
+		<BrowserChrome
+			showAddressBar={!!playground}
+			url={url}
+			toolbarButtons={toolbarButtons}
+			onUrlChange={(url) => playground?.goTo(url)}
+			hideToolbar={hideToolbar}
+			className={className}
+		>
+			<JustViewport />
+		</BrowserChrome>
+	);
+};
 
-interface JustViewportProps {
-	iframeRef: Ref<HTMLIFrameElement>;
-}
+export const JustViewport = function LoadedViewportComponent() {
+	const iframeRef = useRef<HTMLIFrameElement>(null);
+	const iframe = (iframeRef as any)?.current;
+	useEffect(() => {
+		if (iframe) {
+			setupPostMessageRelay(iframe, document.location.origin);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [!!iframe]);
 
-export const JustViewport = function LoadedViewportComponent({
-	iframeRef,
-}: JustViewportProps) {
+	const activeSite = useAppSelector((state) => state.activeSite!);
+	const mountDescriptor = useAppSelector(
+		(state) => state.opfsMountDescriptor
+	);
+
+	const dispatch = useAppDispatch();
 	useEffect(() => {
-		const iframe = (iframeRef! as any).current!;
-		setupPostMessageRelay(iframe, document.location.origin);
+		let unmounted = false;
+		async function doTheWork() {
+			const iframe = (iframeRef as any)?.current;
+			let playground: PlaygroundClient;
+
+			console.log('booting playground');
+
+			try {
+				playground = await bootPlayground({
+					blueprint: activeSite.originalBlueprint!,
```

Comment:
note: I think we may need to build a blueprint from the active site info rather than executing the original blueprint because:
- the site configuration may have deviated from the original blueprint
- the original blueprint may not be idempotent, and creating 100 example posts or clobbering the site DB on every load may easily be undesirable behavior.